lib/aruba/platform.rb

```diff
@@ -182,19 +183,16 @@ def simple_table(hash)
     # rubocop:disable Metrics/MethodLength
     # rubocop:disable Metrics/CyclomaticComplexity
     def which(program, path = ENV['PATH'])
-      on_windows = false
-      on_windows = true if File::ALT_SEPARATOR
-
       program = program.to_s
```

Comment:
Let's move this into separate classes. And use something like `Platform::WhichCommand.new(path).which(program)`. Similar to `EnvironmentVariables` as @mattwynne suggested.

```ruby
def which(program, path = ENV['PATH'])
  Platform::WhichCommand.new(path).which(program)
end
```

I'm not sure where to move the created files best, maybe under `lib/aruba/platform/which_command` and then create another file `lib/aruba/platform/which_command.rb`? Maybe we should also move the `environment.rb`-files as well.

Then create some classes:

```ruby
module Aruba
  class BasicWhichCommand
    # common code
  end
```
```ruby
  class UnixWhichCommand < BasicWhichCommand
    private

    attr_reader :path

    public
    def initialize(path)
      @path = path
    end

    def which(program)
    end

    def self.match?(*)
      true
    end
  end
```
```ruby
  class WindowsWhichCommand < BasicWhichCommand
    private

    attr_reader :path

    public
    def initialize(path)
      @path = path
    end

    def which(program)
    end

    def self.match?(*)
      Platform.windows?
    end
  end
```
```ruby
  class WhichCommand
    def initialize
      @whiches = []
      @whiches = WindowsWhichCommand
      @whiches = UnixWhichCommand

    end

    def which(program, path = ENV['PATH'])
      @which.find { |w| w.match? }.new(path).which(program)
    end
  end
end
```